src/main/java/org/gradle/profiler/studio/invoker/StudioSample.java

```diff
@@ -0,0 +1,34 @@
+package org.gradle.profiler.studio.invoker;
+
+import org.gradle.profiler.result.BuildInvocationResult;
+import org.gradle.profiler.result.Sample;
+
+import java.time.Duration;
+
+public enum StudioSample implements Sample<BuildInvocationResult> {
+    GRADLE_EXECUTION_TIME("Gradle execution time") {
+        @Override
+        public Duration extractFrom(BuildInvocationResult result) {
+            StudioBuildActionResult studioResult = (StudioBuildActionResult) result.getActionResult();
+            return studioResult.getGradleExecutionTime();
+        }
+    },
+    IDE_EXECUTION_TIME("IDE execution time") {
+        @Override
+        public Duration extractFrom(BuildInvocationResult result) {
+            StudioBuildActionResult studioResult = (StudioBuildActionResult) result.getActionResult();
+            return studioResult.getIdeExecutionTime();
+        }
+    };
```

Comment:
I'd move these to `StudioBuildActionResult`, and likewise move `EXECUTION_TIME` to `BuildActionResult`. WDYT?

Reply:
I guess I'm just not very fond of the use of these `enum`s.

BTW, if we do want to use `enum`s, we could change `getName()` to `name()` and that would simplify things significantly.

Reply:
I moved code out of enums. I also created `StudioBuildInvocationResult` and moved samples there. I think it makes sense that they are there since they implement `Sample<StudioBuildInvocationResult>`.